src/modules/previewpane/MonacoPreviewHandler/monaco_languages.json

```diff
@@ -1,961 +1 @@
-        ".cc",
-        ".cxx",
-        ".hpp",
-        ".hh",
-        ".hxx"
-      ],
-      "aliases": [
-        "C++",
-        "Cpp",
-        "cpp"
-      ]
-    },
-    {
-      "id": "csharp",
-      "extensions": [
-        ".cs",
-        ".csx",
-        ".cake"
-      ],
-      "aliases": [
-        "C#",
-        "csharp"
-      ]
-    },
-    {
-      "id": "csp",
-      "extensions": [],
-      "aliases": [
-        "CSP",
-        "csp"
-      ]
-    },
-    {
-      "id": "css",
-      "extensions": [
-        ".css"
-      ],
-      "aliases": [
-        "CSS",
-        "css"
-      ],
-      "mimetypes": [
-        "text/css"
-      ]
-    },
-    {
-      "id": "dart",
-      "extensions": [
-        ".dart"
-      ],
-      "aliases": [
-        "Dart",
-        "dart"
-      ],
-      "mimetypes": [
-        "text/x-dart-source",
-        "text/x-dart"
-      ]
-    },
-    {
-      "id": "dockerfile",
-      "extensions": [
-        ".dockerfile"
-      ],
-      "filenames": [
-        "Dockerfile"
-      ],
-      "aliases": [
-        "Dockerfile"
-      ]
-    },
-    {
-      "id": "ecl",
-      "extensions": [
-        ".ecl"
-      ],
-      "aliases": [
-        "ECL",
-        "Ecl",
-        "ecl"
-      ]
-    },
-    {
-      "id": "fsharp",
-      "extensions": [
-        ".fs",
-        ".fsi",
-        ".ml",
-        ".mli",
-        ".fsx",
-        ".fsscript"
-      ],
-      "aliases": [
-        "F#",
-        "FSharp",
-        "fsharp"
-      ]
-    },
-    {
-      "id": "go",
-      "extensions": [
-        ".go"
-      ],
-      "aliases": [
-        "Go"
-      ]
-    },
-    {
-      "id": "graphql",
-      "extensions": [
-        ".graphql",
-        ".gql"
-      ],
-      "aliases": [
-        "GraphQL",
-        "graphql",
-        "gql"
-      ],
-      "mimetypes": [
-        "application/graphql"
-      ]
-    },
-    {
-      "id": "handlebars",
-      "extensions": [
-        ".handlebars",
-        ".hbs"
-      ],
-      "aliases": [
-        "Handlebars",
-        "handlebars"
-      ],
-      "mimetypes": [
-        "text/x-handlebars-template"
-      ]
-    },
-    {
-      "id": "hcl",
-      "extensions": [
-        ".tf",
-        ".tfvars",
-        ".hcl"
-      ],
-      "aliases": [
-        "Terraform",
-        "tf",
-        "HCL",
-        "hcl"
-      ]
-    },
-    {
-      "id": "html",
-      "extensions": [
-        ".html",
-        ".htm",
-        ".shtml",
-        ".xhtml",
-        ".mdoc",
-        ".jsp",
-        ".asp",
-        ".aspx",
-        ".jshtm"
-      ],
-      "aliases": [
-        "HTML",
-        "htm",
-        "html",
-        "xhtml"
-      ],
-      "mimetypes": [
-        "text/html",
-        "text/x-jshtm",
-        "text/template",
-        "text/ng-template"
-      ]
-    },
-    {
-      "id": "ini",
-      "extensions": [
-        ".ini",
-        ".properties",
-        ".gitconfig"
-      ],
-      "filenames": [
```

Comment:
@Aaron-Junker
Is it possible to have an additional language definition for files that explicit named `setup.ini` and match a special string in the first line?

Reply:
No not directly. You can only register for specific filetypes. What could work is register the handler for ini files and then look how the file starts.

Reply:
What's the usecase for filename then?

Reply:
It's an autogenerated file by monaco, so there is data which we don't need.